webextensions/common/tst-api.js

```diff
@@ -438,7 +443,11 @@ export async function sendMessage(aMessage, aOptions = {}) {
         error: e
       };
     }
-  }));
+  };
+
+  for (const id of aTargetSet) {
+    yield sendMessage(id);
```

Comment:
元のコードではPromise.allでいきなり全てのメッセージを並行送信しているので、この変更だと挙動が変わってしまう気がします（具体的には、遅くなる）。
現在ドキュメントに書いてあるAPIの仕様としては、他のアドオンが登録したリスナに影響を与えるかどうかということは全く規定していません。なので、この現状でリスナの処理だけシーケンシャルに行うようにしてもそれほどメリットがあるようには思えないのですが、どうでしょうか？

むしろ、これを入れるのであれば、DOMのイベントリスナの仕様のように、誰か一人が他の後続リスナによるlistenを阻害できるようにするという仕様をキッチリ盛り込んだAPIに改訂した方がよさそうです。

Reply:
@piroor 

> 元のコードではPromise.allでいきなり全てのメッセージを並行送信しているので、この変更だと挙動が変わってしまう気がします（具体的には、遅くなる）。

I think there is no perf regression such your concern by following reasons:

* In this change, this `sendMessage()` is async function. If we call it without `await`, it would behave like a function which return `Promise` simply.
* This means that `spawnMessages()` spawns to call `sendMessage()` for all items in `aTargetSet` without awaiting each of the result of it sequentially.
* So this change would not cause a perf regression caused by async messaging.

Reply:
なるほど、Task.jsmを使う場面とか色々混同して混乱しまっていました。yieldの時点ではpromiseがすぐに返されて、それで話が終わってるんですね。

最後の部分、`return Array.from(targetSet).map(send)`でもよさそうに見えるんですが、yieldを使う方がよい理由があるということでしょうか？ 後学のために知っておきたいので、ポインタを示していただけると助かります。

Reply:
と自分で書いた後で、Array.from分のコストがかからないということなのだと気がつきました。Promise.allは引数に配列しか受け付けないという先入観があったので。